### PR TITLE
single-instance container (poo#129883)

### PR DESCRIPTION
**Pull request by okurz**
* openqa-bootstrap: Speedup package installation
* openqa-bootstrap: Optionally configure SUSE-specifics and tests initialization
* openqa-bootstrap: Support systemd-less environments
* container: Handle potentially already existing certificates gracefully

### CODE REVIEW

File: script/openqa-bootstrap [hunk 1]
```diff
@@ -26,21 +32,26 @@ fi
 
 
 # install packages
-zypper -n install --no-recommends openQA-single-instance qemu-arm qemu-ppc qemu-x86 qemu-tools sudo iputils os-autoinst-distri-opensuse-deps
+pkgs='openQA-single-instance qemu-arm qemu-ppc qemu-x86 qemu-tools sudo iputils os-autoinst-distri-opensuse-deps'
```
Comment (baierjan) [inline script/openqa-bootstrap:35]:
```suggestion
pkgs=(openQA-single-instance qemu-arm qemu-ppc qemu-x86 qemu-tools sudo iputils os-autoinst-distri-opensuse-deps)
```

File: script/openqa-bootstrap [hunk 2]
```diff
@@ -26,21 +32,26 @@ fi
...
 
 if [ "$(uname -m)" = "aarch64" ]; then
-    zypper -n install --no-recommends qemu-uefi-aarch64
+    pkgs="$pkgs qemu-uefi-aarch64"
```
Comment (baierjan) [inline script/openqa-bootstrap:38]:
```suggestion
    pkgs+=(qemu-uefi-aarch6)
```

File: script/openqa-bootstrap [hunk 3]
```diff
@@ -26,21 +32,26 @@ fi
...
 fi
 
 # this was split into a separate package on newer dist versions - so install it if available
 if zypper -n search -x qemu-hw-display-virtio-gpu-pci ; then
-    zypper -n install --no-recommends qemu-hw-display-virtio-gpu qemu-hw-display-virtio-gpu-pci
+    pkgs="$pkgs qemu-hw-display-virtio-gpu qemu-hw-display-virtio-gpu-pci"
```
Comment (baierjan) [inline script/openqa-bootstrap:43]:
```suggestion
    pkgs+=(qemu-hw-display-virtio-gpu qemu-hw-display-virtio-gpu-pci)
```

File: script/openqa-bootstrap [hunk 4]
```diff
@@ -26,21 +32,26 @@ fi
...
 fi
-
+# shellcheck disable=SC2086
+zypper -n install --no-recommends $pkgs
```
Comment (baierjan) [inline script/openqa-bootstrap:46]:
```suggestion
zypper -n install --no-recommends "${pkgs[@]}"
```

**Review by baierjan — COMMENTED** [review]
AFAIK you can use this to make shellcheck happy

File: script/openqa-bootstrap [hunk 5]
```diff
@@ -26,21 +26,25 @@ fi
...
 
 if [ "$(uname -m)" = "aarch64" ]; then
-    zypper -n install --no-recommends qemu-uefi-aarch64
+    pkgs+=(qemu-uefi-aarch64)
 fi
 
 # this was split into a separate package on newer dist versions - so install it if available
 if zypper -n search -x qemu-hw-display-virtio-gpu-pci ; then
-    zypper -n install --no-recommends qemu-hw-display-virtio-gpu qemu-hw-display-virtio-gpu-pci
+    pkgs=(qemu-hw-display-virtio-gpu qemu-hw-display-virtio-gpu-pci)
```
Comment (asdil12) [inline script/openqa-bootstrap:37]:
```suggestion
    pkgs+=(qemu-hw-display-virtio-gpu qemu-hw-display-virtio-gpu-pci)
```

**Review by kalikiana — APPROVED** [review]

**Review by baierjan — APPROVED** [review]

File: script/openqa-bootstrap [hunk 6]
```diff
@@ -68,24 +72,26 @@ if ping -c1 download.suse.de. && (! rpm -q ca-certificates-suse) ; then
 fi
 
 # fetch tests and needles
-if ping -c1 gitlab.suse.de. ; then
-    # use faster local mirror if run from within SUSE network
-    export needles_giturl="https://gitlab.suse.de/openqa/os-autoinst-needles-opensuse-mirror.git"
-fi
-/usr/share/openqa/script/fetchneedles
```
Comment (baierjan) [inline script/openqa-bootstrap:75]: And suddenly, fetchneedles is not called for opensuse (ie. `$skip_suse_tests` not set), see https://progress.opensuse.org/issues/130201

Comment (Martchus) [reply script/openqa-bootstrap:75]: I read the "And suddenly" as "this is an obvious mistake". So are you going to prepare a PR to execute this non-SUSE-specific unconditionally?

Although I'm wondering where `skip_suse_tests` is actually set to a non-empty value.

**Review by Martchus — COMMENTED** [review]
This PR seems problematic and is indeed likely the cause of https://progress.opensuse.org/issues/130201.

File: script/openqa-bootstrap [hunk 7]
```diff
@@ -5,12 +5,12 @@ set -xeuo pipefail
 
 dbname="${dbname:="openqa"}"
 dbuser="${dbuser:="geekotest"}"
-BYPASS_SYSTEMD_CHECK="${BYPASS_SYSTEMD_CHECK:=""}"
+running_systemd=
+skip_suse_specifics=
+skip_suse_tests=
```
[the remark this answers is not in the document]
Comment (baierjan) [reply script/openqa-bootstrap:10]: Both skip variables are unset and will remain unset during the whole script. If you set them externally, they will be unset anyways. So yes, we should fix that as well.

**Pull request merged**